sql-plugin/src/main/scala/org/apache/spark/sql/execution/datasources/rapids/GpuPartitioningUtils.scala

```diff
@@ -121,7 +118,7 @@ object GpuPartitioningUtils {
     basePathOption match {
       case Some(userDefinedBasePath) =>
         val fs = userDefinedBasePath.getFileSystem(hadoopConf)
-        if (!fs.isDirectory(userDefinedBasePath)) {
+        if (!fs.getFileStatus(userDefinedBasePath).isDirectory) {
```

Comment:
Is it possible to return a `null` from `getFileStatus` ?

Reply:
Throws FileNotFoundException, its javadoc:
    FileNotFoundException – when the path does not exist;